.github/workflows/emulator.yml

```diff
@@ -0,0 +1,134 @@
+        emulator: ${{ fromJson(needs.generate-matrices.outputs.emulator) }}
+    runs-on: ubuntu-22.04
+    env:
+      ndk_version: 26.3.11579264
+      conan_host_profile: android-${{ matrix.emulator.build_api_level }}-${{ matrix.emulator.arch }}
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: setup python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+      - name: install python dependencies and setuptools (required to build GLib)
+        run: pip install --upgrade pip conan setuptools
+
+      - name: install NDK
+        run: echo "y" | sudo ${ANDROID_HOME}/cmdline-tools/latest/bin/sdkmanager --install "ndk;${{ env.ndk_version }}"
+
+      - name: conan remote
+        run: conan remote add --index 0 odr https://artifactory.opendocument.app/artifactory/api/conan/conan
+      - name: conan config
+        run: conan config install .github/config/ubuntu-22.04/conan
+
+      # Multiple unpublished packages in this repo have inter-dependencies
+      # Can't build them until the deps are published to artifactory or at least exported
+      - name: Conan export all packages
+        run: python scripts/conan_export_all_packages.py
+
+      - name: conan install
+        run: conan install ${{ matrix.package.conanfile }} --version ${{ matrix.package.version }} --build missing --profile:host ${{ env.conan_host_profile }} --profile:build ubuntu
+      - name: conan source
+        run: conan source ${{ matrix.package.conanfile }} --version ${{ matrix.package.version }}
+      - name: conan build
+        run: conan build ${{ matrix.package.conanfile }} --version ${{ matrix.package.version }} --profile:host ${{ env.conan_host_profile }} --profile:build ubuntu
+      - name: conan export
+        run: conan export ${{ matrix.package.conanfile }} --version ${{ matrix.package.version }}
+      - name: conan export-pkg
+        run: conan export-pkg ${{ matrix.package.conanfile }} --version ${{ matrix.package.version }} --profile:host ${{ env.conan_host_profile }} --profile:build ubuntu
+
+      - name: Enable KVM group perms
+        run: |
+          echo 'KERNEL=="kvm", GROUP="kvm", MODE="0666", OPTIONS+="static_node=kvm"' | sudo tee /etc/udev/rules.d/99-kvm4all.rules
+          sudo udevadm control --reload-rules
+          sudo udevadm trigger --name-match=kvm
+
+      - uses: reactivecircus/android-emulator-runner@v2
+        with:
+          force-avd-creation: false
+          emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none -no-snapshot-save
+          api-level: ${{ matrix.emulator.emulator_api_level }}
+          arch: ${{ matrix.emulator.arch }}
+          target: ${{ matrix.emulator.api_type_target }}
+          script: conan test ${{ matrix.package.test_conanfile }} ${{ matrix.package.package_reference }} --build missing --profile:host ${{ env.conan_host_profile }} --profile:build ubuntu 
```

Comment:
oh nice you can run the conan test inside the emulator?

Reply:
Ideally pdf2htmlEX test should include an actual conversion, not just object creation, but I've left that as a next step